python/runtime/cudaq/operators/py_handlers.cpp

```diff
@@ -0,0 +1,259 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include <complex>
+#include <pybind11/complex.h>
+#include <pybind11/functional.h>
+#include <pybind11/numpy.h>
+#include <pybind11/operators.h>
+#include <pybind11/stl.h>
+
+#include "cudaq/operators.h"
+#include "py_handlers.h"
+
+namespace cudaq {
+
+void bindPauli(py::module mod) {
+  py::enum_<pauli>(mod, "Pauli",
+                   "An enumeration representing the types of Pauli matrices.")
+      .value("X", pauli::X)
+      .value("Y", pauli::Y)
+      .value("Z", pauli::Z)
+      .value("I", pauli::I);
+}
+
+void bindOperatorHandlers(py::module &mod) {
+  using matrix_callback = std::function<complex_matrix(
+      const std::vector<int64_t> &, const parameter_map &)>;
+
+  auto cmat_to_numpy = [](const complex_matrix &m) {
+    std::vector<ssize_t> shape = {static_cast<ssize_t>(m.rows()),
+                                  static_cast<ssize_t>(m.cols())};
+    std::vector<ssize_t> strides = {
+        static_cast<ssize_t>(sizeof(std::complex<double>) * m.cols()),
+        static_cast<ssize_t>(sizeof(std::complex<double>))};
+
+    // Return a numpy array without copying data
+    return py::array_t<std::complex<double>>(shape, strides, m.data);
+  };
```

Comment:
It might be nice to make this a standalone helper, as this seems to be a helpful function.